clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -708,6 +708,8 @@ public void handleResponse(AbstractResponse response) {
                     return;
                 } else if (error == Errors.TOPIC_AUTHORIZATION_FAILED) {
                     unauthorizedTopics.add(topicPartition.topic());
+                } else if (error == Errors.OPERATION_NOT_ATTEMPTED) {
+                    log.info("{}Did not attempt to add partition {} to transaction because other partitions in the batch had errors.", logPrefix, topicPartition);
```

Comment:
I'm debating if it's worthwhile being more defensive in order to ensure that we cannot reach the non-error path below. For example, we could set a flag indicating that there were unadded partitions and error out if we don't break first. What do you think?

Also, perhaps this can be DEBUG level since we don't know whether the request will just be retried?